File: Android/tuiroomkit/src/main/java/com/tencent/cloud/tuikit/roomkit/model/entity/RoomInfo.java
package com.tencent.cloud.tuikit.roomkit.model.entity;


public class RoomInfo {
    public String  name;
    public String  owner;
    public String  roomId;
    public boolean isOpenCamera;
    public boolean isUseSpeaker;
    public boolean isOpenMicrophone;
    public boolean enableVideo       = true;
    public boolean enableAudio       = true;
    public boolean enableMessage     = true;
    public boolean enableSeatControl = false;

    @Override
    public String toString() {
        return "RoomInfo{"
                + "name='" + name + '\''
                + ", owner='" + owner + '\''
                + ", roomId='" + roomId + '\''
                + ", isOpenCamera=" + isOpenCamera
                + ", isOpenMicrophone=" + isOpenMicrophone
                + ", isUseSpeaker=" + isUseSpeaker
                + ", enableVideo=" + enableVideo
                + ", enableAudio=" + enableAudio
                + ", enableMessage=" + enableMessage
                + ", enableSeatControl=" + enableSeatControl
                + '}';
    }
}

File: iOS/TUIRoomKit/Source/Model/RoomInfo.swift
//
//  RoomInfo.swift
//  TUIRoomKit
//
//  Created by WesleyLei on 2022/9/27.
//

import Foundation
import TUIRoomEngine

public class RoomInfo {
    public var roomId: String {
        set {
            roomInfo.roomId = newValue
        }
        get {
            return roomInfo.roomId
        }
    }

    public var name: String {
        set {
            roomInfo.name = newValue
        }
        get {
            return roomInfo.name
        }
    }

    public var isOpenMicrophone: Bool
    public var isOpenCamera: Bool

    public var enableVideo: Bool {
        set {
            roomInfo.enableVideo = newValue
        }
        get {
            return roomInfo.enableVideo
        }
    }

    public var enableAudio: Bool {
        set {
            roomInfo.enableAudio = newValue
        }
        get {
            return roomInfo.enableAudio
        }
    }

    public var enableMessage: Bool {
        set {
            roomInfo.enableMessage = newValue
        }
        get {
            return roomInfo.enableMessage
        }
    }

    var roomType: TUIRoomType {
        set {
            roomInfo.roomType = newValue
        }
        get {
            return roomInfo.roomType
        }
    }

    public var owner: String {
        return roomInfo.owner
    }

    private var roomInfo: TUIRoomInfo
    public init() {
        isOpenMicrophone = false
        isOpenCamera = false
        roomInfo = TUIRoomInfo()
    }

    public func update(engineRoomInfo: TUIRoomInfo) {
        roomInfo = engineRoomInfo
    }

    public func getEngineRoomInfo() -> TUIRoomInfo {
        return roomInfo
    }
}
